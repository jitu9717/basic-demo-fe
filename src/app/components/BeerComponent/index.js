import React from 'react';
import { PropTypes } from 'prop-types';

const BeerComponent = ({ beerContentDetail }) => {
    return (
         <div className="col-sm-6  col-md-6 col-lg-4 commonBoxModel">
              <div className="beerContainer">
                <h3 className="beertype">{beerContentDetail.beerType}</h3>
              </div>
          </div>
    );
};

BeerComponent.propTypes = {
    beerContentDetail: PropTypes.object.isRequired,
};

export default BeerComponent;