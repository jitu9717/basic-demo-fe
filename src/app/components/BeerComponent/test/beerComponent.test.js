import React from 'react';
import { shallow } from 'enzyme';

import BeerComponent from '../index';

describe('BeerComponent component renders the BeerComponent correctly', () => {
    let beerComp, allProps;

    beforeEach(()=> {
        allProps = {
            beerContentDetail: { beerType: "Aler Beer", id: 1 },
        };
        beerComp = shallow(<BeerComponent {...allProps} />);
    });

    it('render the BeerComponent component', () => {
        expect(beerComp.length).toEqual(1)
    });

    it('renders with no data', () => {
        allProps.beerContentDetail = {};
        expect(beerComp).toMatchSnapshot();
    });
});
