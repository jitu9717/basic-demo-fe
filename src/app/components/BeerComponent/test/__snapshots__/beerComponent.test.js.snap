// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerComponent component renders the BeerComponent correctly renders with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BeerComponent
    beerContentDetail={
        Object {
            "beerType": "Aler Beer",
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="beerContainer"
>
        <h3
                className="beertype"
        >
                Aler Beer
        </h3>
</div>,
      "className": "col-sm-6  col-md-6 col-lg-4 commonBoxModel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="beertype"
>
          Aler Beer
</h3>,
        "className": "beerContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Aler Beer",
          "className": "beertype",
        },
        "ref": null,
        "rendered": "Aler Beer",
        "type": "h3",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="beerContainer"
>
          <h3
                    className="beertype"
          >
                    Aler Beer
          </h3>
</div>,
        "className": "col-sm-6  col-md-6 col-lg-4 commonBoxModel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3
            className="beertype"
>
            Aler Beer
</h3>,
          "className": "beerContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aler Beer",
            "className": "beertype",
          },
          "ref": null,
          "rendered": "Aler Beer",
          "type": "h3",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
